xarray/core/_typed_ops.pyi

```diff
@@ -0,0 +1,731 @@
+"""Stub file for mixin classes with arithmetic operators."""
+# This file was generated using xarray.util.generate_ops. Do not edit manually.
+
+from typing import NoReturn, TypeVar, Union, overload
+
+import numpy as np
+
+from .dataarray import DataArray
+from .dataset import Dataset
+from .groupby import DataArrayGroupBy, DatasetGroupBy, GroupBy
+from .variable import Variable
+
+try:
+    from dask.array import Array as DaskArray
+except ImportError:
+    DaskArray = np.ndarray
+
+T_Dataset = TypeVar("T_Dataset", bound="DatasetOpsMixin")
+T_DataArray = TypeVar("T_DataArray", bound="DataArrayOpsMixin")
+T_Variable = TypeVar("T_Variable", bound="VariableOpsMixin")
+
+Scalar = Union[int, float, complex, bytes, str, np.generic]
+ArrayLike = Union[np.ndarray, DaskArray]
+ScalarOrArray = Union[Scalar, ArrayLike]
+DsCompatible = Union[Dataset, DataArray, Variable, GroupBy, ScalarOrArray]
+DaCompatible = Union[DataArray, Variable, DataArrayGroupBy, ScalarOrArray]
+VarCompatible = Union[Variable, ScalarOrArray]
+GroupByIncompatible = Union[Variable, GroupBy]
```

Comment:
Why is groupby incompatible here?

Reply:
GroupBy can't do `grp + grp` or `grp + var`

Reply:
thanks!